uvtools/dspec.py

```diff
@@ -965,15 +774,24 @@ def dayenu_filter(x, data, wgts, filter_dimensions, filter_centers, filter_half_
         raise ValueError("length of filter_dimensions cannot exceed 2")
     # make sure filter_dimensions are 0 or 1.
     for dim in filter_dimensions:
-        if not isinstance(dim,int):
-            raise ValueError("only integers are valid filter dimensions")
-    # make sure that all filter dimensions are valid for the supplied data.
-    if not np.all(np.abs(np.asarray(filter_dimensions)) < data.ndim):
-        raise ValueError("invalid filter dimensions provided, must be 0 or 1/-1")
+        if not dim in [0, 1] or not isinstance(dim, (int, np.int)):
+            raise ValueError("filter dimension must be integer 0, or 1")
+
     # convert filter dimensions to a list of integers (incase the dimensions were supplied as floats)
     # will only filter each dim a single time.
     # now check validity of other inputs. We perform the same check over multiple
     # inputs by iterating over a list with their names.
+    #convert 1d data to 2d data to save lines of code.
+    if d_dim == 1:
```

Comment:
I keep this functionality in the functions I want to exposed to users. 

My rational for keeping dayenu_filter exposed is that it offers dayenu filtering with a much shorter doc-string so some might find it easier to parse the fourier_filter. 

I decided not to promote the same exposure to fit_basis_2d since there are a bunch of complicated basis dependent choices that I think can be a bit overwhelming.